import "goby/common/protobuf/option_extensions.proto";
import "goby/common/protobuf/app_base_config.proto";

package goby.moos.protobuf;

message MOOSGatewayConfig
{
    optional AppBaseConfig base = 1;

    optional string moos_server_host = 2 [default = "localhost"];
    optional uint32 moos_server_port = 3 [default = 9000];
    optional uint32 moos_comm_tick = 4 [default = 5];
    repeated string moos_subscribe_filter = 5
        [(goby.field).description =
             "Substring filter for all MOOS subscriptions; 'NAV_' matches "
             "'NAV_X', 'NAV_Y', etc.; '' matches everything."];
    repeated string goby_subscribe_filter = 6
        [(goby.field).description =
             "Substring filter for all Goby (ZeroMQ) subscriptions; 'NAV_' "
             "matches 'NAV_X', 'NAV_Y', etc.; '' matches everything."];

    repeated string load_shared_library = 10
        [(goby.field).description =
             "Path to a shared library containing compiled Protobuf files. "
             "Preferred over load_dccl_proto_file."];

    repeated string load_proto_file = 11
        [(goby.field).description =
             "Path to a Protobuf file. Use load_shared_library when possible."];

    message ProtobufMOOSBridgePair
    {
        required string pb_group = 1;
        required string moos_var = 2;
        enum Direction
        {
            PB_TO_MOOS = 1;
            MOOS_TO_PB = 2;
        };
        required Direction direction = 3;
    }
    repeated ProtobufMOOSBridgePair pb_pair = 12
        [(goby.field).description =
             "Convert MOOS Variables to Goby PB Groups and vice/versa based on "
             "these directives."];

    optional bool pb_convert = 13 [
        default = false,
        (goby.field).description =
            "Use a DynamicProtobuf Node on the PB side instead of a MOOSNode."
    ];
}
